### PR TITLE
feat: add a new quit-app helper 

### PR DESCRIPTION
**Pull request by katiamoklokova**
This PR adds a brand new `quit-app` helper which will be used for several upcoming new tests. Additionally, `processes` test is enhanced with the app quitting and making sure no processes are left behind.

### CODE REVIEW

File: src/helpers/quit-app.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  await sendNativeKeyboardEvent({
+    text: 'q',
+    cmdOrCtrl: true
+  });
+  // Wait until we have no processes
+  await wait(500);
+  const runningSlackProcessesCount = await getRunningSlackProcessesCount(
+    appState
+  );
+  return runningSlackProcessesCount;
```
Comment (erickzhao) [inline src/helpers/quit-app.ts:15]: Is there any case where running the `quit()` helper should give more than 0 processes?

Comment (katiamoklokova) [reply src/helpers/quit-app.ts]: Thank you @erickzhao! I updated both files, would that work?

File: src/helpers/quit-app.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  await sendNativeKeyboardEvent({
+    text: 'q',
+    cmdOrCtrl: true
+  });
+  // Wait until we have no processes
+  await wait(500);
+  const runningSlackProcessesCount = await getRunningSlackProcessesCount(
+    appState
+  );
+  runningSlackProcessesCount === 0;
```
Comment (erickzhao) [inline src/helpers/quit-app.ts:15]: Make the user know if the helper has failed to quit all Slack processes by raising an error.
```suggestion
  if (runningSlackProcessesCount > 0) {
    throw new Error('Failed to quit Slack from quit helper!');
  }
```

File: src/smoke/processes.ts [hunk 3]
```diff
@@ -1,34 +1,18 @@
...
-      appState
-    );
-    assert.equal(
-      runningSlackProcessesCount,
-      0,
-      `number of running processes is ${runningSlackProcessesCount}, expected 0.`
-    );
-
-    // Restart the app
+    const runningSlackProcessesCount = await quit();
```
Comment (erickzhao) [inline src/smoke/processes.ts:10]: I think would still want to assert this equality here for the purposes of the test (even though the helper now also checks for the number of processes).
```suggestion
    await quit();
    const runningSlackProcessesCount = await getRunningSlackProcessesCount(	
      appState	
    );	
    assert.equal(	
      runningSlackProcessesCount,	
      0,	
      `number of running processes is ${runningSlackProcessesCount}, expected 0.`	
    );
```

File: src/helpers/quit-app.ts (hunk 1 above)
Comment (erickzhao) [reply src/helpers/quit-app.ts]: @katiamoklokova I left a few suggestions!

Comment (katiamoklokova) [reply src/helpers/quit-app.ts]: Ahh it all makes sense 🙌

**Comment by erickzhao** [comment]
@katiamoklokova I think CI is failing because this PR is coming from a fork and Travis isn't configured for it to work.

Going to merge anyways, since the changes work locally for me. :)

**Pull request merged**